### ISSUE TITLE
BOLT is not printing version when ran with --version in some modes

### ISSUE THREAD

**Issue by rafaelauler** [facebookincubator/BOLT#160]
At some point we regressed, since running BOLT on older versions was printing this:
LLVM (http://llvm.org/):
  LLVM version 7.0.0svn
  Optimized build with assertions.
  Default target: x86_64-unknown-linux-gnu
  Host CPU: broadwell

BOLT revision 2ff448dee7461b1b31ae06837e8d859b77985e6b
  Registered Targets:
    aarch64    - AArch64 (little endian)
    aarch64_be - AArch64 (big endian)
    arm64      - ARM64 (little endian)
    x86        - 32-bit X86: Pentium-Pro and above
    x86-64     - 64-bit X86: EM64T and AMD64

But now, it is printing this:
LLVM (http://llvm.org/):
  LLVM version 13.0.0git
  Optimized build with assertions.
  Default target: x86_64-unknown-linux-gnu
  Host CPU: broadwell

**Comment by Sameeranjoshi** [comment]
Here's what I see, I am on `rebased` branch pulled from `5e8e189e201dab1fb5ccada2f4c44e7011f938cc`.
Is the expected patch for this issues to show all the architectures (aarch64* missing ones) given that I see X86 as mentioned below?
```
$./build/bin/llvm-bolt --version
LLVM (http://llvm.org/):
  LLVM version 13.0.0git
  Optimized build with assertions.
  Default target: x86_64-unknown-linux-gnu
  Host CPU: znver1
BOLT revision 5e8e189e201dab1fb5ccada2f4c44e7011f938cc
  Registered Targets:
    x86    - 32-bit X86: Pentium-Pro and above
    x86-64 - 64-bit X86: EM64T and AMD64

$ ./build/bin/llvm-boltdiff --version
LLVM (http://llvm.org/):
  LLVM version 13.0.0git
  Optimized build with assertions.
  Default target: x86_64-unknown-linux-gnu
  Host CPU: znver1
```

**Comment by rafaelauler** [comment]
Thanks @Sameeranjoshi for checking this, looks like we may or may not print the revision depending on how we are called. So I'll rephrase this to just print the BOLT revision in all modes.

**Comment by rafaelauler** [comment]
I don't think we need to print the targets everywhere, but printing the revision is very important.
